File: linkedlist/linkedlist.js


function listNode(value){
    this.val = value;
    this.next = null
  }
  
  var p = new listNode(10)
  p.next = new listNode(20)
  p.next.next = new listNode(2)
  p.next.next.next = new listNode(6)
  
  var linkedlist = function (head){
    // var runner = head;
    // var count = 0;
  
    if(!head){
      head = this.val
    } else {
      runner = head;
    while(runner.next){
      runner = runner.next;
      // count++;
    } 
    runner.next = this.val
    }
    return head
  }

  var printLinkedList = function(head) {
      var t = head
      while(t) {
          console.log(t.val)
          console.log("---")
          t = t.next
      }
  }
  
  console.log(linkedlist(p))
//   console.log(printLinkedList(p))

File: linkedlist/linkedlist.py
class Node(object):
  def __init__(self, val):
    self.val = val
    self.next = None
  
class LinkedList(object):
  def __init__(self):
    self.head = None

  def add(self, val):
    newnode = Node(val)
    newnode.next = self.head
    self.head = newnode
  
  def printLinkedList(self):
    temp = self.head
    print "LinkedList - "
    while temp is not None:
      print temp.val
      temp = temp.next


if __name__ == "__main__":
  print "welcome to python"
  list = LinkedList()
  list.printLinkedList()
  list.add(5)
  list.printLinkedList()
  list.add(6)
  list.printLinkedList()